src/main/webapp/app/tutorialgroup/overview/course-tutorial-group-detail/course-tutorial-group-detail.component.ts

```diff
@@ -1,70 +1,259 @@
-                    next: ({ courseResult, tutorialGroupResult }) => {
-                        this.tutorialGroup = tutorialGroupResult.body ?? undefined;
-                        this.course = courseResult.body ?? undefined;
-                    },
-                });
+    private router = inject(Router);
+    private locale = toSignal(this.translateService.onLangChange.pipe(map((event) => event.lang)), {
+        initialValue: this.translateService.currentLang,
+    });
+    private averageAttendanceRatio = computed<number | undefined>(() => this.computeAverageAttendanceRation(this.tutorialGroup().sessions, this.tutorialGroup().capacity));
+
+    course = input.required<Course>();
+    tutorialGroup = input.required<TutorialGroupDetailGroupDTO>();
+    tutorialGroupSessions = computed<TutorialGroupDetailSession[]>(() => {
+        this.locale();
+        return this.computeSessionsToDisplay(this.selectedSessionListOption(), this.tutorialGroup().sessions);
+    });
+    nextSession = computed<TutorialGroupDetailSession | undefined>(() => this.computeNextSessionDataUsing(this.tutorialGroup().sessions));
+    teachingAssistantImageUrl = computed(() => addPublicFilePrefix(this.tutorialGroup().teachingAssistantImageUrl));
+    tutorialGroupLanguage = computed<string>(() => this.tutorialGroup().language);
+    tutorialGroupCapacity = computed<string>(() => String(this.tutorialGroup().capacity ?? '-'));
+    tutorialGroupMode = computed<string>(() => (this.tutorialGroup().isOnline ? 'artemisApp.generic.online' : 'artemisApp.generic.offline'));
+    tutorialGroupCampus = computed<string>(() => this.tutorialGroup().campus ?? '-');
+    averageAttendancePercentage = computed<string | undefined>(() => this.computeAverageAttendancePercentage(this.averageAttendanceRatio()));
+    pieChart = viewChild(PieChartComponent);
+    pieChartData = computed<NgxChartsSingleSeriesDataEntry[]>(() => this.computePieChartData(this.averageAttendanceRatio()));
+    pieChartColors = computed<Color>(() => this.computePieChartColor(this.averageAttendanceRatio()));
+    sessionListOptions = computed(() => {
+        this.locale();
+        return this.computeSessionListOptions();
+    });
+    selectedSessionListOption = signal<ListOption>('all-sessions');
+    messagingEnabled = computed<boolean>(() => isMessagingEnabled(this.course()));
+    tutorChatLink = computed(() => this.computeTutorChatLink(this.course().id, this.tutorialGroup().tutorChatId));
+    groupChannelLink = computed(() => this.computeGroupChannelLink(this.course().id, this.tutorialGroup().groupChannelId));
+
+    readonly faFlag = faFlag;
+    readonly faUsers = faUsers;
+    readonly faTag = faTag;
+    readonly faCircleExclamation = faCircleExclamation;
+    readonly faCalendar = faCalendar;
+    readonly faClock = faClock;
+    readonly faMapPin = faMapPin;
+    readonly faBuildingColumns = faBuildingColumns;
+    readonly faQuestion = faQuestion;
+    readonly faCheck = faCheck;
```

Comment:
```suggestion
    protected readonly faFlag = faFlag;
    protected readonly faUsers = faUsers;
    protected readonly faTag = faTag;
    protected readonly faCircleExclamation = faCircleExclamation;
    protected readonly faCalendar = faCalendar;
    protected readonly faClock = faClock;
    protected readonly faMapPin = faMapPin;
    protected readonly faBuildingColumns = faBuildingColumns;
    protected readonly faQuestion = faQuestion;
    protected readonly faCheck = faCheck;
   ```
  and please move them to the top, so they can easily be skipped when reviewing/reading the file